fuel-tx/src/transaction/types/input/message.rs

```diff
@@ -143,15 +183,20 @@ pub mod specifications {
 #[derive(Default, Derivative, Clone, PartialEq, Eq, Hash)]
 #[derivative(Debug)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "da-compression",
+    derive(fuel_compression::Compress, fuel_compression::Decompress)
+)]
 #[derive(fuel_types::canonical::Deserialize, fuel_types::canonical::Serialize)]
 pub struct Message<Specification>
 where
-    Specification: MessageSpecification,
+    Specification: MessageSpecification + Clone,
 {
     /// The sender from the L1 chain.
     pub sender: Address,
     /// The receiver on the `Fuel` chain.
     pub recipient: Address,
+    #[cfg_attr(feature = "da-compression", da_compress(skip))] // Stored on L1
     pub amount: Word,
```

Comment:
I think we can skip `sender`, `recipient`, `amount` and `data` since it is available via event form L1

Reply:
Added skips in d23d6cd3c5d78ad0115aab5df39627902cfab374